service/collector/nat.go

```diff
@@ -0,0 +1,271 @@
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+
+	clientaws "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/aws-operator/service/controller/key"
+	"github.com/giantswarm/aws-operator/service/internal/cache"
+)
+
+const (
+	// __NATCache__ is used as temporal cache key to save NAT response.
+	prefixNATcacheKey = "__NATCache__"
+	labelVPC          = "vpc"
+	labelAZ           = "availability_zone"
+)
+
+const (
+	subsystemNAT = "nat"
+)
+
+var (
+	natDesc *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystemNAT, "info"),
+		"NAT limits information.",
+		[]string{
+			labelAccountID,
+			labelVPC,
+			labelAZ,
+		},
+		nil,
+	)
+)
+
+type NATConfig struct {
+	Helper *helper
+	Logger micrologger.Logger
+
+	InstallationName string
+}
+
+type NAT struct {
+	cache  *natCache
+	helper *helper
+	logger micrologger.Logger
+
+	installationName string
+}
+
+type natCache struct {
+	cache *cache.StringCache
+}
+
+type natInfoResponse struct {
+	Vpcs map[string]vpcInfo
+}
+
+type vpcInfo struct {
+	NatGatewaysByZone map[string]float64
+}
+
+func NewNAT(config NATConfig) (*NAT, error) {
+	if config.Helper == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Helper must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.InstallationName == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.InstallationName must not be empty", config)
+	}
+
+	v := &NAT{
+		// AWS operator creates at this moment one NAT for each private subnet (node
+		// pool). As clusters are not created nor changed so often, and the process
+		// can take around 20 minutes, 30 minutes for the cache expiration is a
+		// reasonable value.
+		cache:  newNATCache(time.Minute * 30),
+		helper: config.Helper,
+		logger: config.Logger,
+
+		installationName: config.InstallationName,
+	}
+
+	return v, nil
+}
+
+func newNATCache(expiration time.Duration) *natCache {
+	cache := &natCache{
+		cache: cache.NewStringCache(expiration),
+	}
+
+	return cache
+}
+
+func (n *natCache) Get(accID string) (*natInfoResponse, error) {
+	var c natInfoResponse
+	raw, ok := n.cache.Get(prefixNATcacheKey + accID)
+	if ok {
```

Comment:
```suggestion
	raw, exists := n.cache.Get(prefixNATcacheKey + accID)
	if exists {
```
Minor note but in general naming that boolean as suggested helps with readability. When quickly glancing through the code it's easier to pick up what's happening :slightly_smiling_face: